septentrion/cli.py

```diff
@@ -5,32 +5,54 @@
 """
 import logging
 import os
-from typing import Any, Dict
+from typing import Any, TextIO
 
 import click
 from click.types import StringParamType
 
-from septentrion import __version__, core, db, migrate, settings, style, utils
+from septentrion import (
+    __version__,
+    configuration,
+    core,
+    db,
+    exceptions,
+    migrate,
+    style,
+    utils,
+)
 
 logger = logging.getLogger(__name__)
 
 
-def get_context_settings() -> Dict[str, Any]:
+def load_config(ctx: click.Context, param: click.Parameter, value: TextIO) -> None:
+    if not value:
+        try:
+            file_contents, file = configuration.read_default_configuration_files()
+        except exceptions.NoDefaultConfiguration:
+            pass
+    else:
+        file = getattr(value, "name", "stdin")
+        logger.info(f"Reading configuration from {file}")
+        file_contents = value.read()
+
     try:
-        with open("septentrion.ini") as file:
-            config = file.read()
-    except FileNotFoundError:
-        config = ""
+        default = configuration.parse_configuration_file(file_contents)
+    except exceptions.NoSeptentrionSection:
+        if file in configuration.CONFIGURATION_FILES:
+            click.echo(
+                f"Configuration file found at {file} but contains no septentrion "
+                "section"
+            )
```

Comment:
I find this mildly annoying, because it will show up everytime you launch septentrion if you have a `setup.cfg` without a `[septentrion]` section in it (and no other conf file of course). I understand the intent though.

Reply:
Note that it is already the current behavior, nothing changes here.

Reply:
> it will show up everytime you launch septentrion

Hm, maybe I screw things up, but I remember writing this so that it would only output the message if it found a septentrion.ini. I'll check.

> Note that it is already the current behavior, nothing changes here.

Note really, before this PR, only septentrion.ini was read.

Reply:
Ok :+1: .